habitat_sim/simulator.py

```diff
@@ -145,6 +145,21 @@ def _config_pathfinder(self, config: Configuration):
                 f"Could not find navmesh {navmesh_filenname}, no collision checking will be done"
             )
 
+        agent_legacy_config = AgentConfiguration()
+        default_agent_config = config.agents[config.sim_cfg.default_agent_id]
+        if not np.isclose(
+            agent_legacy_config.radius, default_agent_config.radius
+        ) or not np.isclose(agent_legacy_config.height, default_agent_config.height):
+            logger.info(
+                f"Recomputing navmesh for agent's height {default_agent_config.height} and radius"
+                f" {default_agent_config.radius}."
+            )
+            navmesh_settings = NavMeshSettings()
+            navmesh_settings.set_defaults()
+            navmesh_settings.agent_radius = default_agent_config.radius
+            navmesh_settings.agent_height = default_agent_config.height
+            self.recompute_navmesh(self.pathfinder, navmesh_settings)
```

Comment:
Does recompute_navmesh work with blind agents (when the mesh isn't loaded into memory)? I have encountered some errors in the past under that scenario. Should we throw a more informative error for blind agents or throw a warning and use the incorrect navmesh?